### PR TITLE
Test on PHP 8

### PR DESCRIPTION
**Pull request by j0k3r**
Also:
- use Composer v2 in tests

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -39,7 +39,10 @@
         "guzzlehttp/guzzle": "^6.0"
     },
     "require-dev": {
-        "friendsofphp/php-cs-fixer": "^2.0",
-        "symfony/phpunit-bridge": "^5.0"
+        "friendsofphp/php-cs-fixer": "^3.0",
+        "symfony/phpunit-bridge": "^6.0"
```
Comment (peter279k) [inline composer.json:43]: This dependency will require `php-7.1` version at least. And it should have the dependency issue when running `composer install` command.

Comment (j0k3r) [reply composer.json]: Yes, thanks for checking, I reverted back!

File: lib/Imgur/Auth/OAuth2.php [hunk 2]
```diff
@@ -12,8 +12,8 @@
  */
 class OAuth2 implements AuthInterface
 {
-    const AUTHORIZATION_ENDPOINT = 'https://api.imgur.com/oauth2/authorize';
-    const ACCESS_TOKEN_ENDPOINT = 'https://api.imgur.com/oauth2/token';
+    public const AUTHORIZATION_ENDPOINT = 'https://api.imgur.com/oauth2/authorize';
+    public const ACCESS_TOKEN_ENDPOINT = 'https://api.imgur.com/oauth2/token';
```
Comment (peter279k) [inline lib/Imgur/Auth/OAuth2.php]: This syntax is only available on `php-7.1` at least.

Comment (peter279k) [reply lib/Imgur/Auth/OAuth2.php]: Please look at the [Example #4](https://www.php.net/manual/en/language.oop5.constants.php).

Comment (j0k3r) [reply lib/Imgur/Auth/OAuth2.php]: Yep already reverted

**Pull request merged**